### DISCUSSION TITLE
Unique contraint across multiple fields

### DISCUSSION THREAD

**Discussion by FlixMa** [pocketbase/pocketbase#543] [Q&A] [answered]
Hey!

My schema contains a junction table as discussed [here](https://github.com/pocketbase/pocketbase/discussions/289#discussion-4295210).
However, I would like to restrict the creation of multiple `response` records belong to the same User and Event.
Ticking the unique property of both the user and the event relation unfortunately restricts the use of duplicate values individually.
Usually I would create a unique constraint consisting of the tuple (user_id, event_id) but this does not seem to be possible, right?

How would you suggest to go about it?

**Comment by benallfree** [comment]
This could work https://github.com/pocketbase/pocketbase/discussions/542#discussioncomment-3675856

I haven't tested this, but something like this might work:

```go
app.Dao().DB().NewQuery(`
    CREATE UNIQUE INDEX your_unique_index_name ON YOUR_COLLECTION_NAME (uid) WHERE (...some select count * query...) = 0;
`).Execute();
```

**Comment by ganigeorgiev** [comment, marked as answer]
> Usually I would create a unique constraint consisting of the tuple (user_id, event_id) but this does not seem to be possible, right?

Currently only unique constraints over a single field can be defined from the Admin UI.

As a workaround, for now you can define your multi-fields constraint manually or via migration by executing something like:

```go
app.Dao().DB().NewQuery(`
    CREATE UNIQUE INDEX responses_user_id_event_id_idx ON responses (user_id, event_id);
`).Execute();
```

**Reply by FlixMa** [reply]
With the proposed change it didn't work right away. That could be my fault however, as I just noticed that you specified this to work only during migration.
Anyway, I was able create the constraint manually by using an sqlite db editor. Now the API returns a 400 error code and the addition of the duplicate is prevented.
Thanks!
